src/entity.ts

```diff
@@ -78,9 +78,11 @@ export class Entity {
             entityId: bindingInfo.self,
             operationName: currentRequest.name,
             isNewlyConstructed: !batchState.entityExists,
-            getState: this.getState.bind(this, batchState),
+            getState: this.getState.bind(this, batchState) as <T>(

```

Comment:
I must admit that I'm struggling to read the `getState` signature. Do you mind describing it?

Reply:
Ah, thanks for the playground , I see it it now.
I think this solution makes sense for now, and perhaps we'll revisit the type signature and its associated refactoring in the future, assuming a cleaner syntax is possible.

Reply:
You might be able to make a function yourself that does the currying, but then you're probably going to lose the performance optimisations that the JS runtimes have around how `Function.prototype.bind` works.